packages/fast-components-react-base/src/listbox-item/listbox-item.tsx

```diff
@@ -0,0 +1,172 @@
+                className={this.generateClassNames()}
+                role="option"
+                id={this.props.id}
+                aria-selected={isSelected}
+                aria-disabled={this.props.disabled}
+                onClick={this.handleClick}
+                onFocus={this.handleFocus}
+                onKeyDown={this.handleKeyDown}
+            >
+                {this.props.displayString}
+                {this.props.children}
+            </div>
+        );
+    }
+
+    /**
+     * Create class-names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.listboxItem", "");
+
+        if (this.props.disabled) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.listboxItem__disabled",
+                ""
+            )}`;
+        }
+
+        return super.generateClassNames(classNames);
+    }
+
+    /**
+     * Invoke this option
+     */
+    private invokeOption(
+        event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>
+    ): void {
+        const itemData: ListboxItemData = {
+            id: this.props.id,
+            value: this.props.value,
+            displayString: this.props.displayString,
+        };
+
+        if ((this.context as ListboxContextType).itemInvoked) {
+            (this.context as ListboxContextType).itemInvoked(itemData, event);
+        }
+
+        if (typeof this.props.onInvoke === "function") {
+            this.props.onInvoke(event, this.props);
+        }
+    }
+
+    /**
+     * Handle the keydown event of the item
+     */
+    private handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>): void => {
+        if (this.props.disabled) {
+            return;
+        }
+
+        switch (e.keyCode) {
+            case KeyCodes.enter:
+            case KeyCodes.space:
+                e.preventDefault();
+                this.invokeOption(e);
+                break;
+        }
+
+        if (typeof this.props.onKeyDown === "function") {
+            this.props.onKeyDown(e);
+        }
+    };
+
+    /**
+     * Handle the keydown event of the item
+     */
+    private handleClick = (e: React.MouseEvent<HTMLDivElement>): void => {
+        if (this.props.disabled) {
+            return;
+        }
+
+        e.preventDefault();
+        this.invokeOption(e);
+
+        if (typeof this.props.onClick === "function") {
+            this.props.onClick(e);
+        }
+    };
+
+    /**
+     * Handle focus event
+     */
+    private handleFocus = (e: React.FocusEvent<HTMLDivElement>): void => {
+        if (this.props.disabled) {
+            return;
+        }
+
+        const itemData: ListboxItemData = {
+            id: this.props.id,
+            value: this.props.value,
+            displayString: this.props.displayString,
+        };
+
+        if (typeof (this.context as ListboxContextType).itemFocused === "function") {
```

Comment:
If what you are saying is that different context types can collide then we definitely should prefix with component name.  How about:
    listboxItemInvoked
    listboxItemInvoked
    listboxSelectedItems
    listboxMultiselectable

Reply:
If what you are saying is that different context types can collide then we definitely should prefix with component name. How about:
listboxItemInvoked
listboxItemInvoked
listboxSelectedItems
listboxMultiselectable

Reply:
Since it is a method I think we normally prefix it with "on" on this case, because you would be calling it correct? That's why my suggestion would be `onListboxItemFocus`, `onListboxItemInvoke`, same with the others if they are methods